assembly/__tests__/roundtrip.spec.ts

```diff
@@ -0,0 +1,109 @@
+import { JSONDecoder } from "../decoder";
+import { JSONEncoder } from "../encoder";
+
+let handler: JSONEncoder;
+let decoder: JSONDecoder<JSONEncoder>;
+
+function roundripTest(jsonString: string, expectedString: string | null = null): bool {
+  log<string>("--------" + jsonString + (expectedString ? " " + expectedString : ""));
+  expectedString = expectedString || jsonString;
+  let buffer: Uint8Array = new Uint8Array(jsonString.lengthUTF8);
+  let utf8ptr = jsonString.toUTF8();
+  memory.copy(buffer.buffer.data, utf8ptr, buffer.byteLength);
+  decoder.deserialize(buffer);
+  let resultBuffer = handler.serialize();
+  let resultString = String.fromUTF8(
+    resultBuffer.buffer.data,
+    resultBuffer.length
+  );
+  expect<string>(resultString).toStrictEqual(expectedString);
+  expect<string>(handler.toString()).toStrictEqual(expectedString);
+  return true;
+}
+
+describe("Round trip", () => {
+
+  beforeEach(() => {
+    handler = new JSONEncoder();
+    decoder = new JSONDecoder<JSONEncoder>(handler);
+  });
+
+
+  it("create Decoder", () => {
```

Comment:
nit: somewhat weird to capitalize this way, can be `"create decoder"`

Reply:
Ah yes, this was because I just did a copy paste of your old tests where this was camelCase.  I'll make them all lower case